### PR TITLE
OOP#1 Homework

### PR DESCRIPTION
**Pull request by NG90**
Authors: @NG90 @kozlofski @kyrtaq @Maciej3206 @cnuebred @DarkAriesX

### CODE REVIEW

File: shm/player.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    std::shared_ptr<Ship> getShip() const { return ship_; };
+    int getMoney() const { return money_; };
+    size_t getAvailableSpace() const { return availableSpace_; };
+
+    size_t getSpeed() const { return ship_->getSpeed(); };
+    Cargo* getCargo(size_t index) const { return ship_->getCargo(index); };
+
+private:
+    std::shared_ptr<Ship> ship_{};
```
Comment (majkel84) [inline shm/player.hpp:20]: consider using alias for std::shared_ptr<Ship>

File: shm/ship.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#include <string>
+#include <vector>
+
+#include "cargo.hpp"
+
+class Ship {
+public:
+    Ship();
+    Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id);
+    Ship(int maxCrew, int speed, size_t id);
```
Comment (majkel84) [inline shm/ship.hpp:13]: try to keep same order in constructor

**Review by Kasperice — APPROVED** [review]
Looks good for me

File: shm/map.hpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include <vector>
```
Comment (WBigaj) [inline shm/map.hpp:3]: There should be newline between #pragma once and vector

**Review by WBigaj — COMMENTED** [review]
I have checked your code. Looks solid since I haven't found everything barring single oversight.

File: shm/island.hpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+class Island {
+public:
+    class Coordinates {
+    public:
+        Coordinates() = default;
+        Coordinates(int positionX, int positionY);
+        bool operator==(const Coordinates& rhs) const;
+
+    private:
+        int positionX_{};
```
Comment (DamiXioR) [inline shm/island.hpp:12]: I think you don't need initialize class variables.

File: shm/map.cpp [hunk 5]
```diff
@@ -0,0 +1,45 @@
+#include "map.hpp"
+
+#include <algorithm>
+#include <random>
+#include <vector>
+
+constexpr size_t totalIslands = 10;
```
Comment (DamiXioR) [inline shm/map.cpp:7]: It's better to use upper cases

**Review by DamiXioR — COMMENTED** [review]
Good job

File: shm/map.cpp [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    for (size_t i = 0; i < howMany; i++) {
+        int newX;
+        int newY;
+        do {
+            newX = xDistrib(gen);
+            newY = yDistrib(gen);
+        } while (std::any_of(islands_.begin(), islands_.end(), [=](auto island) {
+            return island.getPosition() == Island::Coordinates(newX, newY);
+        }));
+        islands_.emplace_back(Island(newX, newY));
```
Comment (nauka-programowania-MA) [inline shm/map.cpp]: when you use empalce back, dont create temporary object. call constructor directly. `emplace_back(newX, newY)`

File: shm/ship.hpp (hunk 2 above)
Comment (nauka-programowania-MA) [reply shm/ship.hpp:13]: +1

**Review by nauka-programowania-MA — APPROVED** [review]

**Comment by nauka-programowania-MA** [comment]
{
"requirement_ids": [2568, 2568, 2568, 2568, 2568, 2568, 2568, 2568, 2569, 2570],
"points": [5, 5, 5, 5, 5, 5, 5, 5, 20, 6],
"users": ["NG90", "kyrtaq", "kozlofski ", "Maciej3206 ", "cnuebred", "DarkAriesX"]
}

**Pull request closed without merging**